include/envoy/stats/stats.h

```diff
@@ -24,6 +24,32 @@ class Instance;
 
 namespace Stats {
 
+/**
+ * Struct stored under Server::Options to hold information about the maximum object name length and
+ * maximum stat suffix length of a stat. These have defaults in StatsOptionsImpl, and the maximum
+ * object name length can be overridden. The default initialization is used in IsolatedStatImpl, and
+ * the user-overridden struct is stored in Options.
+ */
+class StatsOptions {
+public:
+  virtual ~StatsOptions() {}
+
+  /**
+   * The max allowed length of a complete stat name, including suffix.
+   */
+  virtual size_t maxNameLength() const PURE;
+
+  /**
+   * The max allowed length of the object part of a stat name.
```

Comment:
What is the object part and the suffix part of a stat? Can you potentially provide an example in the comments?

Reply:
There's more color in `/source/common/stats/stats_impl.h`, which I'm not sure how much of to duplicate in the include file. So the object part is something like `cluster.<cluster_name>.outlier_detection`, and the suffix part is one of the statistics listed in the docs [here](https://www.envoyproxy.io/docs/envoy/latest/configuration/cluster_manager/cluster_stats) like `ejections_enforced_consecutive_gateway_failure`. We just want to limit the lengths of these halves separately. Do you think it's worth adding that to the include too?

Reply:
Maybe just comment about where the other comments are? In general I would expect the public includes to have the detail so I can understand what things mean, but fine to link to other places. Whatever works. It's mostly that this nomenclature caught me off guard so I think it might confuse others also.

Reply:
That sounds good to me. I did some work in https://github.com/envoyproxy/envoy/pull/3629/commits/732af4f5bc925d78c46c4796ef6f32483d75932c.